### PR TITLE
Fixed #35987 -- Made ErrorList.copy() copy the renderer attribute.

### PR DESCRIPTION
**Pull request by adamchainz**
#### Trac ticket number

ticket-35987

#### Branch description

Fix the issue by adding an extra attribute copy in `ErrorList.copy()`.

I discovered the issue is due to the underlying `UserList.copy()` method in Python, which makes a copy by reinstantiating the class:

https://github.com/python/cpython/blob/2041a95e68ebf6d13f867e214ada28affa830669/Lib/collections/__init__.py#L1334-L1335

That's why there an explicit copy of `error_class` was already added in 49275c548887769cd70bbd85a3b125491f0c4062.

I added tests, both directly for `ErrorList` and `form.get_context()` where an error list is copied. I also added a defensive test for `ErrorDict` to ensure the same issue doesn't occur there, just in ase.

#### Checklist
- [x] This PR targets the `main` branch. 
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [x] I have checked the "Has patch" ticket flag in the Trac system.
- [x] I have added or updated relevant tests.
- [x] I have added or updated relevant docs, including release notes if applicable.
- [x] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

File: tests/forms_tests/tests/test_utils.py
Comment (sarahboyce) [inline tests/forms_tests/tests/test_utils.py:1]: The new tests added here pass without the changes - are they needed? Would make sense to pull them into a different commit maybe

Comment (adamchainz) [reply tests/forms_tests/tests/test_utils.py:1]: Ah, good point. `test_error_list_copy_attributes` was meant to detect the error, but it didn't work because it called `__copy__` instead of `copy()`. I have adjusted it then split the other tests into an extra commit.

File: tests/forms_tests/tests/test_utils.py [hunk 1]
```diff
@@ -161,6 +162,17 @@ def __str__(self):
             '<a href="http://www.example.com/">example</a></li></ul>',
         )
 
+    def test_error_list_copy_attributes(self):
```
Comment (sarahboyce) [inline tests/forms_tests/tests/test_utils.py:183]: This test also passes without the changes :thinking:

Comment (sarahboyce) [reply tests/forms_tests/tests/test_utils.py:183]: Ah sorry! Thank you :star:

**Review by sarahboyce — APPROVED** [review]
Thank you!
